*** compiler.vim	Fri Feb 20 11:13:14 2009
--- mycompiler.vim	Fri Feb 20 14:21:01 2009
***************
*** 320,325 ****
--- 320,330 ----
  "           gvim --servername LATEX --remote-silent +%l "%f"
  "      For inverse search, if you are reading this, then just pressing \ls
  "      will work.
+ "
+ " Use DDE to communicate with SumatraPDF, hence calling a Python
+ " file with the DDE gestion. Use same s:path as used for
+ " calling outline.py in texviewer.vim
+ let s:path = expand('<sfile>:p:h')
  function! Tex_ForwardSearchLaTeX()
  	if &ft != 'tex'
  		echo "calling Tex_ForwardSeachLaTeX from a non-tex file"
***************
*** 342,351 ****
  	
  	" inverse search tips taken from Dimitri Antoniou's tip and Benji Fisher's
  	" tips on vim.sf.net (vim.sf.net tip #225)
! 	if (has('win32') && (viewer == "yap" || viewer == "YAP" || viewer == "Yap"))
  
! 		let execString = 'silent! !start '. viewer.' -s '.line('.').expand('%').' '.mainfnameRoot
  
  
  	elseif (has('macunix') && (viewer == "Skim" || viewer == "PDFView" || viewer == "TeXniscope"))
  		" We're on a Mac using a traditional Mac viewer
--- 347,366 ----
  	
  	" inverse search tips taken from Dimitri Antoniou's tip and Benji Fisher's
  	" tips on vim.sf.net (vim.sf.net tip #225)
! 	if (has('win32') && (viewer == "yap" || viewer == "YAP" || viewer == "Yap" || viewer =~? "^sumatrapdf"))
! 
! 		if (viewer == "yap" || viewer == "YAP" || viewer == "Yap")
  
! 			let execString = 'silent! !start '. viewer.' -s '.line('.').expand('%').' '.mainfnameRoot
  
+ 		elseif viewer =~? "^sumatrapdf"
+ 			
+ 			" SumatraPDF requires the .tex file path to be given relatively to
+ 			" the main file path, hence the substitute
+ 			let relativeFile=substitute(expand("%:p"), Tex_GetMainFileName(':p:h').'/', '','')
+ 			let execString = 'silent! !'.s:path.'/fwdsumatra.py "'.mainfnameFull.'.'.s:target.'" "'.relativeFile.'" '. line('.')
+ 
+ 		endif
  
  	elseif (has('macunix') && (viewer == "Skim" || viewer == "PDFView" || viewer == "TeXniscope"))
  		" We're on a Mac using a traditional Mac viewer
